### PR TITLE
TST: Parametrize in pandas/tests/internals/test_internals.py

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
Yea looks like a nice cleanup

File: pandas/tests/internals/test_internals.py [hunk 1]
```diff
@@ -334,13 +334,9 @@ def test_pickle(self, mgr):
...
         assert not mgr2._known_consolidated
 
-    def test_non_unique_pickle(self):
-
-        mgr = create_mgr("a,a,a:f8")
-        mgr2 = tm.round_trip_pickle(mgr)
-        tm.assert_frame_equal(DataFrame(mgr), DataFrame(mgr2))
-
-        mgr = create_mgr("a: f8; a: i8")
+    @pytest.mark.parametrize("mgr_string", ["a,a,a:f8", "a: f8; a: i8"])
```
Comment (jbrockmendel) [inline pandas/tests/internals/test_internals.py]: are the spaces meaningful?

File: pandas/tests/internals/test_internals.py [hunk 2]
```diff
@@ -541,54 +540,41 @@ def _compare(old_mgr, new_mgr):
...
+    )
+    def test_interleave_dtype(self, mgr_string, dtype):
+        mgr = create_mgr(mgr_string.format(dtype=dtype))
+        assert mgr.as_array().dtype == dtype
 
+    @pytest.mark.parametrize(
+        "mgr_string, dtype",
+        [
+            ("a: category", "i8"),
+            ("a: category; b: category", "i8"),
```
Comment (jbrockmendel) [inline pandas/tests/internals/test_internals.py]: heads up: this case is incorrect, as is the one on L558.  xref #32519.

**Comment by jreback** [comment]
can you rebase

**Comment by jbrockmendel** [comment]
##[error]./pandas/tests/internals/test_internals.py:558:5:F811:redefinition of unused 'test_interleave' from line 549

**Review by jreback — APPROVED** [review]
lgtm. @jbrockmendel merge when ok.

**Comment by jbrockmendel** [comment]
@MomIsBestFriend can you rebase and ping on green

**Comment by MomIsBestFriend** [comment]
Restarting azure as I am not sure why it failed.

**Comment by MomIsBestFriend** [comment]
ping @jbrockmendel, the CI is green.

**Comment by jbrockmendel** [comment]
thanks @MomIsBestFriend

**Pull request merged**